### PR TITLE
Return early if user has site-wide permission and non strict access

### PR DESCRIPTION
**Pull request by amitaibu**
#70

### CODE REVIEW

**Comment by amitaibu** [comment]
Going over the code showed me we are still node centric. I see two issues we need to figure out:

`$entity_access_strict = $config->get('node_access_strict');` For simplicity we should change it to `entity_access_strict`, although it might be needed to be per-entity. don't know yet, but for no I'll change it.

Also, currently in the PR we're doing `$account->hasPermission("create $entity_type_id content")` -- but that again is a node specific permission. If it would be a different entity type can we get the site-wide "create" permission?

**Comment by chx** [comment]
Based on a discussion with @berdir, there is no way to do this cross entity wise. I would check for entity type id node (most common) and do a shortcut for node and forget the rest.

File: og.module [hunk 1]
```diff
@@ -125,7 +125,7 @@ function og_entity_access(EntityInterface $entity, $operation, AccountInterface
   $entity_type_id = $entity->getEntityTypeId();
   $bundle_id = $entity->bundle();
 
-  $entity_access_strict = \Drupal::config('og.settings')->get('node_access_strict');
+  $entity_access_strict = \Drupal::config('og.settings')->get('entity_access_strict');
```
Comment (damiankloip) [inline og.module]: This setting was already converted too. So this is all that's needed :+1:

File: og.module [hunk 2]
```diff
@@ -171,17 +171,14 @@ function og_entity_create_access(AccountInterface $account, array $context, $bun
...
-  // Save some legwork if the user has the core permission and strict node
-  // access is not set.
-
   $config = \Drupal::config('og.settings');
-  $entity_access_strict = $config->get('node_access_strict');
-  // @todo: $entity isn't passed.
-  /*
-  if (!$entity_access_strict && $entity->access('create', $account)) {
+  $entity_access_strict = $config->get('entity_access_strict');
+
```
Comment (damiankloip) [inline og.module:180]: We might need to think about some plugins or other way to declare this for different entity types maybe?

**Comment by amitaibu** [comment]
Yes, I agree with @chx - the underlying API is entity agnostic, but the _real_ usage of OG will remain mostly nodes. So PR now converts the variable back to `node_access_strict`. I believe this approach, at least for now, is good enough.

**Comment by amitaibu** [comment]
this still requires tests

**Comment by chx** [comment]
Tests incoming.

**Comment by chx** [comment]
Erm, `nope, og_entity_access()` testing should be a followup, assign it to me. It needs a "framework" like the one for the OgAccess tests and is way too big for this issue.

**Comment by amitaibu** [comment]
Sure, so I'll set this one as needs review, and create a follow-up.

**Pull request merged**